app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapView.jsx

```diff
@@ -415,6 +416,20 @@ class SamplesHeatmapView extends React.Component {
       this.fetchMetadataFieldsBySampleIds(),
     ]);
 
+    const pipelineMajorVersionsSet = new Set(
+      compact(
+        map(data => {
+          if (!data.pipeline_version) return null;
+          return `${data.pipeline_version.split(".")[0]}.x`;
+        }, heatmapData)
+      )
+    );
+    if (pipelineMajorVersionsSet.size > 1) {
```

Comment:
It looks like this notification is shown over and over every time the user changes their filters, even if they are still viewing the same samples. Perhaps we should only show the notification once per component lifecycle?

Also, a nit that perhaps this logic should be separated out into a function `checkPipelineVersions()`.